lib/ruby_lsp/requests/code_lens.rb

```diff
@@ -44,17 +45,52 @@ def run
       def visit_class(node)
         class_name = node.constant.constant.value
         if class_name.end_with?("Test")
-          add_code_lens(node, [@path, class_name, BASE_COMMAND + " -Itest " + @path])
+          add_code_lens(node, path: @path, name: class_name, command: BASE_COMMAND + " -Itest " + @path)
+        end
+        visit(node.bodystmt)
+      end
+
+      sig { override.params(node: SyntaxTree::DefNode).void }
+      def visit_def(node)
+        if @modifier == "public"
+          method_name = node.name.value
+          if method_name.start_with?("test")
+            add_code_lens(
+              node,
+              path: @path + ":" + node.location.start_line.to_s,
```

Comment:
The `:` technique won't work without Rails. We need to use `TESTOPTS="-n='/NAME_OF_TEST/'"` to select which one to run unfortunately.

Also, it seems like we have both `--name` and the line number. We should probably only use one. I think `path:` here is just `@path` and command is
```ruby
TESTOPTS="-n='/NAME_OF_TEST$/'" + BASE_COMMAND ...
```

Reply:
When I run something like:
```
TESTOPTS="-n='/test_initialize_enabled_features_with_array$/'" bundle exec ruby -Itest <PATH_TO_TEST>/test/executor_test.rb
```
It runs the entire test suite instead of the individual test.

Reply:
Sorry, it should actually just be the `--name` parameter. It accepts a complete string, which is what I believe we should use to avoid matching other tests.
```shell
bundle exec ruby -Itest <PATH_TO_TEST>/test/executor_test.rb --name test_initialize_enabled_features_with_array
```